### ISSUE TITLE
Environment variables in write_relabel_configs 

### ISSUE THREAD

**Issue by pro100guy** [prometheus/prometheus#10582]
# Proposal
There was a huge discussion related to environment variables in Prometheus already https://github.com/prometheus/prometheus/issues/2357 and as a result, Env var is currently available in the external_labels entry.
It would be great to have something similar in "remote_write.write_relabel_configs" section.

## Use case: 
2 Prometheus servers running in HA.  Many other components scraping the metrics from them (like Alertmanager, etc).
I want to add a new label to the metrics which are sent to remote_write endpoint (to split them on the remote side only), for example __replica__ (can be hostname). "external_labels" can be used, but then metrics also will be divided into 2 groups for components like Alertnamager, which mean 2 times more alerts.

## Suggestion: 
Add possibility to use env variables in  "write_relabel_configs":

## Dummy example:
```
...
global:
  ...
  external_labels:
    __replica__: all
...
remote_write:
- url: http://remote_url/push
  write_relabel_configs:
  - separator: ; 
    regex: (.*)
    target_label: __replica__
    replacement: ${POD_NAME}
    action: replace
...
```

Where **POD_NAME** can be taken from the environment:
```
   env:
   - name: POD_NAME
     valueFrom:
       fieldRef:
         fieldPath: metadata.name
```

So, basically, I set external_labels the same for all Prometheus servers and in Alertmanager we will see only 1 alert. 
But in remote_write section, we rewrote the value  __replica__ and sent relabeled metrics to the remote server.

**Comment by roidelapluie** [comment]
You could do:

```
global:
  ...
  external_labels:
    __replica__: $POD_NAME

alerting:
  - alert_relabel_configs:
    - target_label: __replica__
      replacement: __all__
```


The issue with relabel configs is that `$` is also used for regexes, so it is not an option to have env subst here.

**Comment by pro100guy** [comment]
Thanks @roidelapluie 
It was exactly what I was looking for
